gocourse17/cmd/server/main_test.go

```diff
@@ -0,0 +1,215 @@
+package main
+
+import (
+	"context"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/vicuani/go_course/gocourse17/internal/grpc/grpcapi"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/grpc/test/bufconn"
+)
+
+const bufSize = 1024 * 1024
+
+func startTestServer(t *testing.T) (dialer func(context.Context, string) (net.Conn, error)) {
+	lis := bufconn.Listen(bufSize)
+
+	baseServer := grpc.NewServer()
+	grpcapi.RegisterTaxiServiceServer(baseServer, newServer())
+	go func() {
+		if err := baseServer.Serve(lis); err != nil {
+			t.Errorf("error serving server: %v", err)
+		}
+	}()
+
+	t.Cleanup(func() {
+		err := lis.Close()
+		if err != nil {
+			t.Fatalf("error closing listener: %v", err)
+		}
+		baseServer.Stop()
+	})
+
+	return func(ctx context.Context, _ string) (net.Conn, error) { return lis.DialContext(ctx) }
+}
+
+func newTestClient(t *testing.T, ctx context.Context, dialer func(context.Context, string) (net.Conn, error)) grpcapi.TaxiServiceClient {
+	conn, err := grpc.DialContext(ctx, "", grpc.WithContextDialer(dialer), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		t.Fatalf("error connecting to server: %v", err)
+	}
+	t.Cleanup(func() {
+		if err := conn.Close(); err != nil {
+			t.Fatalf("error closing connection: %v", err)
+		}
+	})
+
+	return grpcapi.NewTaxiServiceClient(conn)
+}
+
+func TestEvaluateCargoState(t *testing.T) {
+	dialer := startTestServer(t)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+	client := newTestClient(t, ctx, dialer)
+
+	t.Run("incorrect driver id", func(t *testing.T) {
+		req := &grpcapi.EvaluateCargoStateRequest{
+			DriverId:   -4,
+			CargoState: grpcapi.Rating_RATING_EXCELLENT,
+		}
+
+		_, err := client.EvaluateCargoState(ctx, req)
+		if err == nil {
+			t.Fatalf("Expected error 'incorrect driver id' but got no error")
+		}
```

Comment:
Для тестів gRPC рекомендовано перевіряти ще код помилки через функцію [`status.Code(err)`](https://pkg.go.dev/google.golang.org/grpc/status#Code)


```suggestion
		if code := status.Code(err); code != codes.InvalidArgument {
			t.Errorf("Expected status code %v, got %v", codes.InvalidArgument, code)
		}
```

Якщо `err == nil`, то це означає `codes.OK`.